Recruit-Training/Embedded-Recruit-Training/Core/Src/main.c

```diff
@@ -288,6 +328,41 @@ void HAL_CAN_RxFifo0MsgPendingCallback(CAN_HandleTypeDef* hcan)
     }
 
     //TODO: Match StdId of header and data length content for green and blue messages and check data for set bit of toggling green and blue led
+    if ((hdr.StdId == BLUE_MESSAGE_STDID || hdr.StdId == GREEN_MESSAGE_STDID) && hdr.DLC == 1)
+    {
+        if (data[0] == 1 && data[3] == 1 && data[7] == 1)
+        {
+            if (BlueToggleFlag)
+                BlueToggleFlag = 0;
+            else
+                BlueToggleFlag = 1;
+
+        }
+    
+        if (data[0] == 1 && data[1] == 1)
```

Comment:
Same thing down here. You can probably get rid of whatever you're doing with a for loop down there when you implement the bitmask correctly. 

Each value of data is 8 bits since it's `uint8_t`. So `data[0]`'s value is 8 bits long. I think you're thinking like if I have a message of `0101` it becomes `data = {0, 1, 0, 1}`, but it's actually like `data={0000 0101, blah, blah}`. 

Correct me if I'm wrong...